examples/kprobetcp/kprobetcp-ebpf/src/main.rs

```diff
@@ -0,0 +1,86 @@
+#![no_std]
+#![no_main]
+
+#[allow(non_upper_case_globals)]
+#[allow(non_snake_case)]
+#[allow(non_camel_case_types)]
+#[allow(dead_code)]
+mod binding;
+
+use crate::binding::{sock, sock_common};
+
+use aya_bpf::{
+    helpers::bpf_probe_read_kernel, macros::kprobe, programs::ProbeContext,
+};
+
+use aya_log_ebpf::info;
+const AF_INET: u16 = 2;
+const AF_INET6: u16 = 10;
+
+#[kprobe(name = "kprobetcp")]
+pub fn kprobetcp(ctx: ProbeContext) -> u32 {
+    match try_kprobetcp(ctx) {
+        Ok(ret) => ret,
+        Err(ret) => match ret.try_into() {
+            Ok(rt) => rt,
+            Err(_) => 1,
+        },
+    }
+}
+
+fn try_kprobetcp(ctx: ProbeContext) -> Result<u32, i64> {
+    let sock: *mut sock = ctx.arg(0).ok_or(1i64)?;
+    let sock_com = unsafe {
+        bpf_probe_read_kernel(&(*sock).__sk_common as *const sock_common)
+            .map_err(|e| e)?
+    };
+    // info!(&ctx, "tcp_connect: inet_family: {}", sock_com.skc_family);
+    match sock_com.skc_family {
+        AF_INET => {
+            let src_addr = u32::from_be(unsafe {
+                sock_com.__bindgen_anon_1.__bindgen_anon_1.skc_rcv_saddr
+            });
+            let dest_addr: u32 = u32::from_be(unsafe {
+                sock_com.__bindgen_anon_1.__bindgen_anon_1.skc_daddr
+            });
+            info!(
+                &ctx,
+                "AF_INET src address: {:ipv4}, dest address: {:ipv4}",
+                src_addr,
+                dest_addr,
+            );
+            Ok(0)
+        }
+        AF_INET6 => {
+            let src_addr = sock_com.skc_v6_rcv_saddr;
+            let dest_addr = sock_com.skc_v6_daddr;
+            info!(
+                &ctx,
+                "AF_INET6 src addr: {:x}:{:x}:{:x}:{:x}:{:x}:{:x}:{:x}:{:x}, dest addr: {:x}:{:x}:{:x}:{:x}:{:x}:{:x}:{:x}:{:x}",
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[0]) },
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[1]) },
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[2]) },
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[3]) },
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[4]) },
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[5]) },
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[6]) },
+                unsafe { u16::from_be(src_addr.in6_u.u6_addr16[7]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[0]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[1]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[2]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[3]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[4]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[5]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[6]) },
+                unsafe { u16::from_be(dest_addr.in6_u.u6_addr16[7]) }
+            );
```

Comment:
We have `{:ipv6}` format hint for `[u8; 16]` :) And also, when you use `[u8; 16]` representation of IPv6, you don't have to care about endianess.

```suggestion
            info!(
                &ctx,
                "AF_INET6 src addr: {:ipv6}, dest addr: {:ipv6}",
                src_addr.in6_u.u6_addr8,
                dest_addr.in6_u.u6_addr8,
            );
```

Reply:
changed in [bbc14f5](https://github.com/aya-rs/book/pull/81/commits/bbc14f58e8a41908686b42315b441534dacddd8d)